class.jetpack.php

```diff
@@ -3188,14 +3189,90 @@ public static function get_activation_source( $referer_url ) {
 	/**
 	 * Runs before bumping version numbers up to a new version
 	 *
-	 * @param  string $version    Version:timestamp
-	 * @param  string $old_version Old Version:timestamp or false if not set yet.
-	 * @return null              [description]
+	 * @param string $version    Version:timestamp.
+	 * @param string $old_version Old Version:timestamp or false if not set yet.
 	 */
 	public static function do_version_bump( $version, $old_version ) {
-		if ( ! $old_version ) { // For new sites
-			// There used to be stuff here, but this seems like it might  be useful to someone in the future...
+		if ( $old_version ) { // For existing Jetpack installations.
+			self::send_update_modal_data();
+		}
+	}
+
+	/**
+	 * Prepares the release post content and image data and saves it in the
+	 * state array. This data is used to create the update modal.
+	 */
+	public static function send_update_modal_data() {
+		$post_data = self::get_release_post_data();
+
+		if ( ! isset( $post_data ) || ! isset( $post_data['posts'] ) || ! isset( $post_data['posts'][0] ) ) {
+			return;
+		}
+
+		$post_content = isset( $post_data['posts'][0]['content'] ) ? $post_data['posts'][0]['content'] : null;
+		if ( empty( $post_content ) ) {
+			return;
+		}
+
+		$post_image_src = isset( $post_data['posts'][0]['featured_image'] ) ? $post_data['posts'][0]['featured_image'] : null;
+		$post_link      = isset( $post_data['posts'][0]['URL'] ) ? $post_data['posts'][0]['URL'] : null;
```

Comment:
Do you think we should bail entirely when we can't find any post link? Or should we take that into account in `renderInnerContent` and not render any `ExternalLink` for that link when it doesn't exist?

Reply:
I think we should just bail if we can't find the post link. I made this change in commit 4d71be6.